Boto3 support

briantroy opened the pull request:
Added boto3 support to Rigor. Rigor now attempts to load boto3 and if it can not loads boto2. All tests in test/test_s3.py pass. test/s3.py was modified as needed to generate a proper mock object.

Method: 
I created simple private shadow methods in BotoS3Client to keep things simple. Each public method calls the private shadow method if boto3 is loaded.

briantroy committed:
Updated rigor s3 class to support both boto and boto3. This modification prefers boto3 and falls back to boto.

briantroy committed:
Minor refactor

briantroy committed:
Minor code cleanup.

kaolin commented on lib/s3.py:
```diff
@@ -2,8 +2,17 @@
 from io import BytesIO
-from boto.s3.connection import S3Connection
-from boto.s3.key import Key
+import sys
```
minor issues if you want to address them: we prefer to import only those things we're using (so not all of boto3, not all of sys)

squidpickles replied:
(I would generally import `sys`, because it contains functions and constants; I reserve partial imports for modules containing specific objects I want to use)

kaolin commented on lib/s3.py:
```diff
@@ -2,8 +2,17 @@
+
+
+
+import botocore
```
I believe botocore is only in boto3?

kaolin commented on lib/s3.py:
```diff
@@ -71,15 +80,28 @@ class BotoS3Client(RigorS3Client):
-			connection_args.append(config.get(credentials, 'aws_secret_access_key'))
-		self._conn = S3Connection(*connection_args)
-		self.bucket = self._conn.get_bucket(bucket)
+		if 'boto3' in sys.modules:
```
I feel like having this check only in init and using that to store the result is less likely to lead to potential inconsistencies (i.e. boto is used at first, then some other library pulls in boto3 and all of a sudden the same rigors3 client is doing boto3 things—a strange runtime effect).

Arguably it might even be cleaner to use the init to monkeypatch the shadow methods over the original ones, removing the conditionals from the original methods.

kaolin reviewed (changes requested):
Only required change on my end is moving the import botocore into the try block (unless I'm mistaken); the rest are open for discussion/consideration. :)

briantroy commented:
Don't merge it yet. Found a bug and working on another potential issue while implementing rigor end to end.
I'll create a new pull request once I sort it out.
Brian



Brian Roy | 415.996.8507 | https://briantroy.com
On Tue, May 2, 2017 at 11:10 AM, Kevin R <notifications@github.com> wrote:
@squidpickles commented on this pull request.




In lib/s3.py [https://github.com/blindsightcorp/rigor/pull/2#discussion_r114386061] :

> @@ -2,8 +2,17 @@
 
 from abc import ABCMeta, abstractmethod
 from io import BytesIO
-from boto.s3.connection import S3Connection
-from boto.s3.key import Key
+import sys


(I would generally import sys , because it contains functions and constants; I reserve partial imports for modules containing specific objects I want to use)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub [https://github.com/blindsightcorp/rigor/pull/2#discussion_r114386061] , or mute the thread [https://github.com/notifications/unsubscribe-auth/AADH5A2_VFoZpoz52nPVhef3aubV1gKYks5r13F5gaJpZM4NK1uW] .

briantroy committed:
Fix for boto3 - avoiding a top level folder with no name.

briantroy committed:
Minimizing imports from boto3 and botocore.

briantroy committed:
Made the boto version check only happen in init. Added method to return the version of both being used. Added a test to cover the boto version method.

briantroy commented:
I've limited the import scope for boto3 and botocore, moved the import of botocore into the try block. I've also make the sys.modules check only happen once and added a boto version property to the object. Also added a method to return the current boto version and a new test to cover that method.

hyandell commented:
This repository has been archived. I recommend you work with a forked version of the project at https://github.com/rigorgt/rigor to get this improvement added.

I'm going to close this as it's a new feature/enhancement rather than a bugfix that existing users might find useful to still be able to discover.
